gnovm/pkg/gnolang/op_expressions.go

```diff
@@ -194,8 +194,12 @@ func (m *Machine) doOpRef() {
 			nv.Value = rv2
 		}
 	}
+	elt := xv.TV.T
+	if elt == DataByteType {
```

Comment:
When x := [4]byte{}

This can be represented as .Data or .List type.
.Data type means ArrayValue{Data:[4]byte{}}
.List type means ArrayValue{List:[4]TypedValue}.

Much complexity due to interactions between these, but this allows for optimized []byte logic so we want to keep it. 

x1 := x[1] ; x2p := &x[2];...
x1's type is Int
x2's type is *Int. No biggie.

x[1] = x1
For the purpose of assignment, first we get &x[1], and it looks like PointerValue{TV:[DataByteType, DataByteValue]} (see GetPointerAtIndexInt2). But this DataTypeByte is only for assignment via pointer, it isn't supposed to linger around. So this PR fixes the bug by resolving it here.

Reply:
This is amazing.